internal/v1/handler_oscap.go

```diff
@@ -181,150 +182,439 @@ func (h *Handlers) GetOscapCustomizations(ctx echo.Context, distribution Distrib
+	}
+
+	if bpBody.Openscap == nil {
+		return []BlueprintLintItem{}, []BlueprintLintItem{}, nil
+	}
+
+	compl, err := bpBody.Openscap.AsOpenSCAPCompliance()
+	if err != nil || compl.PolicyId == uuid.Nil {
+		return []BlueprintLintItem{}, []BlueprintLintItem{}, nil
+	}
+
 	d, err := h.server.getDistro(ctx, distro)
 	if err != nil {
-		return nil, err
+		return nil, nil, err
 	}
 	major, minor, err := d.RHELMajorMinor()
 	if err != nil {
-		return nil, err
+		return nil, nil, err
 	}
-	bp, err := h.server.complianceClient.PolicyCustomizations(ctx.Request().Context(), major, minor, policy)
+
+	var allErrors []BlueprintLintItem
+	var allWarnings []BlueprintLintItem
+
+	policyBP, err := h.server.complianceClient.PolicyCustomizations(ctx.Request().Context(), major, minor, compl.PolicyId.String())
 	if err == compliance.ErrorTailoringNotFound {
-		return nil, err
+		allErrors = append(allErrors, BlueprintLintItem{
+			Name:        "Compliance",
+			Description: "Compliance policy does not have a definition for the latest minor version",
+		})
+		policyBP = nil
 	} else if err != nil {
-		return nil, echo.NewHTTPError(http.StatusInternalServerError, err)
+		return nil, nil, err
 	}
 
-	// make sure all packages are present, all partitions, all enabled/disabled services, all kernel args
-	for _, pkg := range bp.GetPackagesEx(false) {
-		if cust.Packages == nil || !slices.Contains(*cust.Packages, pkg) {
-			lintErrors = append(lintErrors, BlueprintLintItem{
-				Name:        "Compliance",
-				Description: fmt.Sprintf("package %s required by policy is not present", pkg),
-			})
-			if fixup {
-				cust.Packages = common.ToPtr(append(common.FromPtr(cust.Packages), pkg))
-			}
-		}
+	// Collect errors and warnings from all lint functions (they now return instead of mutating)
+	errs := lintPackagesE(policyBP, bpBody, fixup)
+	allErrors = append(allErrors, errs...)
+	warns := lintPackagesW(policyBP, snapshotCust)
+	allWarnings = append(allWarnings, warns...)
+	errs = lintFilesystemsE(policyBP, bpBody, fixup)
+	allErrors = append(allErrors, errs...)
+	warns = lintFilesystemsW(policyBP, snapshotCust)
+	allWarnings = append(allWarnings, warns...)
+	errs = lintServicesE(policyBP, bpBody, fixup)
+	allErrors = append(allErrors, errs...)
+	warns = lintServicesW(policyBP, snapshotCust)
+	allWarnings = append(allWarnings, warns...)
+	errs = lintKernelE(policyBP, bpBody, fixup)
+	allErrors = append(allErrors, errs...)
+	warns = lintKernelW(policyBP, snapshotCust)
+	allWarnings = append(allWarnings, warns...)
+	errs = lintFIPSE(policyBP, bpBody, fixup)
+	allErrors = append(allErrors, errs...)
+	warns = lintFIPSW(policyBP, snapshotCust)
+	allWarnings = append(allWarnings, warns...)
+
+	// If we couldn't find the policy and fixup was requested, return error after processing warnings
+	if policyBP == nil && fixup {
+		return nil, nil, echo.NewHTTPError(http.StatusNotFound, compliance.ErrorTailoringNotFound)
```

Comment:
This shouldn't be 404.

Reply:
Welll if the policy got deleted maybe? Though it would have failed before this point in that case.